dashboard/test/models/user_test.rb

```diff
@@ -1686,27 +1686,26 @@ def track_progress(user_id, script_level, result, pairings: nil)
   end
 
   test 'admin_changed? equates nil and false' do
-    user = create :user
-    admin = create :admin
-
-    # no change to admin, false
-    refute user.admin_changed?
-    refute admin.admin_changed?
-
-    user.admin = false
-    refute user.admin_changed?
-
-    user.admin = nil
-    refute user.admin_changed?
-
-    user.admin = true
-    assert user.admin_changed?
+    # admins must be teacher
+    teacher = create :teacher
 
-    admin.admin = nil
-    assert admin.admin_changed?
+    matrix = [
+      [nil, nil, false],
+      [nil, false, false],
+      [nil, true, true],
+      [false, nil, false],
+      [false, false, false],
+      [false, true, true],
+      [true, nil, true],
+      [true, false, true],
+      [true, true, false]
+    ]
 
-    admin.admin = false
-    assert admin.admin_changed?
+    matrix.each do |from, to, result|
+      teacher.update!(admin: from)
+      teacher.admin = to
+      assert_equal result, teacher.admin_changed?
+    end
```

Comment:
LGTM. Mind adding a comment documenting matrix, so that the user doesn't have to sort out its meaning from the code that follows it?